internal/service/ssm/parameter_data_source.go

```diff
@@ -73,10 +73,15 @@ func dataParameterRead(ctx context.Context, d *schema.ResourceData, meta interfa
 
 	d.SetId(aws.StringValue(param.Name))
 	d.Set("arn", param.ARN)
-	d.Set("insecure_value", param.Value)
+	if aws.StringValue(param.Type) == ssm.ParameterTypeSecureString {
```

Comment:
**issue:** I like the idea ! However the condition should be inverted. Since we want to have it in `insecure_value` when it is **not** a  ParameterTypeSecureString.
```
if aws.StringValue(param.Type) != ssm.ParameterTypeSecureString
```

Reply:
Hi Yes! That was my mistake - I've changed it now!